lib/workers/qemu.js

```diff
@@ -16,12 +16,21 @@
 
 'use strict'
 
+const _ = require('lodash')
 const childProcess = require('child_process')
 
-const childKillEventHandler = (child, event) => {
-  process.on(event, (code) => {
+const manageHandlers = (child, register) => {
+  const action = register ? 'once' : 'removeListener'
```

Comment:
Are you sure `removeListener` will work here? The first time you call this with `register: true`, you will create an anonymous function and register it as a handler for the `exit` event. Next time you call the function using `register: false`, a *new* anonymous function will be created, whose address is not the same as the address of the first function, and therefore the old handler will not get removed

Reply:
I guess you can factor away the listener function outside the function, so then you always keep a reference to it, or something like that. Maybe even worth a comment explaining why.